pyproject.toml

```diff
@@ -1,143 +1,167 @@
+  "Programming Language :: Python :: 3.12",
+  "Programming Language :: Python :: 3.13",
+  "Topic :: Games/Entertainment",
+  "Topic :: Software Development :: Libraries :: Python Modules",
+  "Topic :: System :: Monitoring",
+  "Typing :: Typed",
 ]
-packages = [
-    { include = "mcstatus" },
-    { include = "protocol", from = "mcstatus" },
+keywords = ["minecraft", "protocol"]
+requires-python = ">=3.9"
+dependencies = ["asyncio-dgram>=2.1.2", "dnspython>=2.4.2"]
+
+[project.urls]
+Documentation = "https://mcstatus.readthedocs.io"
+"Source code" = "https://github.com/py-mine/mcstatus"
+
+[tool.uv]
+default-groups = ["dev", "lint", "test", "docs"]
+
+[dependency-groups]
+dev = ["poethepoet>=0.32.1"]
+lint = [
+  "pre-commit>=3.3.3",
+  "ruff>=0.9.2",
+  "pyright>=1.1.322",
+  "typing-extensions>=4.7.1",
 ]
+test = [
+  "pytest>=7.4,<9.0",
+  "pytest-asyncio>=0.23.2,<0.26.0",
+  "pytest-cov>=4.1,<7.0",
+  "pytest-rerunfailures>=13,<16",
+  "coverage>=7.3.0",
+]
+docs = [
+  "sphinx>=7.1.2",
+  "sphinx-autodoc-typehints>=1.24,<3.0",
+  "furo>=2022.12.7",
+  "m2r2>=0.3.3",
+  "tomli>=2.0.1",
+  "packaging>=24.2",
+  "docutils>=0.18.1,<0.21",              # temporary fix for https://github.com/CrossNox/m2r2/issues/68
+]
+
+[tool.poe.tasks]
+_lint_ruff = "ruff check --show-source ."
+_lint_pyright = "pyright ."
 
-[tool.poetry.dependencies]
-python = ">=3.9,<4"
-asyncio-dgram = "^2.1.2"
-dnspython = "^2.4.2"
-
-[tool.poetry.group.dev.dependencies]
-pre-commit = ">=3.3.3,<5.0.0"
-tox = "^3.28.0"
-tox-poetry = "0.5.0" # deprecated, no sense in `^`
-
-[tool.poetry.group.test.dependencies]
-pytest = ">=7.4,<9.0"
-pytest-asyncio = ">=0.23.2,<0.26.0"
-pytest-cov = ">=4.1,<7.0"
-pytest-rerunfailures = ">=13,<16"
-coverage = "^7.3.0"
-
-[tool.poetry.group.lint.dependencies]
-ruff = "0.9.2"
-coverage = "^7.3.0"
-pyright = "^1.1.322"
-typing-extensions = "^4.7.1"
-
-[tool.poetry.group.docs]
-optional = true
-
-[tool.poetry.group.docs.dependencies]
-sphinx = "^7.1.2"
-sphinx-autodoc-typehints = ">=1.24,<3.0"
-furo = ">=2022.12.7"
-m2r2 = "^0.3.3"
-tomli = { version = "^2.0.1", markers = "python_version < '3.11'" }
-packaging = "24.2"
-poetry-dynamic-versioning = "*"  # actual version is in `release` group
-
-[tool.poetry.group.release]
-optional = true
-
-[tool.poetry.group.release.dependencies]
-twine = ">=4.0.2,<7.0.0"
-poetry-dynamic-versioning = "^1.1.0"
-
-[tool.poetry.group.workflow-tox]
-optional = true
-
-[tool.poetry.group.workflow-tox.dependencies]
-tox-gh-actions = "^2.12.0"
-poetry = ">=1.5.1,<3.0.0"
-
-[tool.poetry.scripts]
-mcstatus = 'mcstatus.__main__:main'
+[tool.poe.tasks.build]
+cmd = "uv build"
+help = "Builds the whl and tar.gz distributions"
+
+[tool.poe.tasks.docs]
+cmd = "make -C docs/ html"
+help = "Generates documentation locally in html format"
+
+[tool.poe.tasks.format]
+cmd = "ruff format"
+help = "Runs automatic formatting tools"
+
+[tool.poe.tasks.lint]
+sequence = ["_lint_pyright", "_lint_ruff"]
```

Comment:
Ruff crashes with

```
error: unexpected argument '--show-source' found

  tip: a similar argument exists: '--show-files'
```

Reply:
Oh, guess that option got removed, I just copied this one from Kevin's PR

Reply:
fixed in https://github.com/py-mine/mcstatus/pull/950/commits/8622e2b6800e637fa9104f30f11e95f09eb37472